### PR TITLE
TypeScript: Added remaining material definitions.

### PR DESCRIPTION
**Pull request by Mugen87**
Title says it all^^

@Methuselah96 Do you mind verifying the PR? 😇

### CODE REVIEW

File: src/materials/MeshMatcapMaterial.d.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Color } from './../math/Color';
+import { Texture } from './../textures/Texture';
+import { Vector2 } from './../math/Vector2';
+import { MaterialParameters, Material } from './Material';
+
+export interface MeshMatcapMaterialParameters extends MaterialParameters {
+
+  color?: Color | string | number;
+	matMap?: Texture;
+	map?: Texture;
```
Comment (Methuselah96) [inline src/materials/MeshMatcapMaterial.d.ts]: The indentation looks inconsistent.

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

File: src/materials/MeshDepthMaterial.d.ts [hunk 2]
```diff
@@ -2,16 +2,23 @@ import { DepthPackingStrategies } from '../constants';
 import { MaterialParameters, Material } from './Material';
 
 export interface MeshDepthMaterialParameters extends MaterialParameters {
+	depthPacking?: DepthPackingStrategies;
+	displacementMap?: Texture;
```
Comment (Methuselah96) [inline src/materials/MeshDepthMaterial.d.ts]: `Texture` needs to be imported.

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

File: src/materials/MeshDistanceMaterial.d.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { MaterialParameters, Material } from './Material';
+import { Vector3 } from './../math/Vector3';
+import { Texture } from './../textures/Texture';
+
+export interface MeshDistanceMaterialParameters extends MaterialParameters {
+	referencePosition?:  Vector3;
```
Comment (Methuselah96) [inline src/materials/MeshDistanceMaterial.d.ts]: The extra space before `Vector3` should be removed.

File: src/materials/MeshDistanceMaterial.d.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	farDistance?: number;
+	displacementMap?: Texture;
+	displacementScale?: number;
+	displacementBias?: number;
+}
+
+export class MeshDistanceMaterial extends Material {
+  constructor(parameters?: MeshDistanceMaterialParameters);
+
+	referencePosition:  Vector3;
```
Comment (Methuselah96) [inline src/materials/MeshDistanceMaterial.d.ts]: The extra space before `Vector3` should be removed.

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

File: src/materials/MeshDistanceMaterial.d.ts (hunk 4 above)
Comment (Methuselah96) [reply src/materials/MeshDistanceMaterial.d.ts]:
```suggestion
	referencePosition: Vector3;
```

File: src/materials/MeshDistanceMaterial.d.ts (hunk 3 above)
Comment (Methuselah96) [reply src/materials/MeshDistanceMaterial.d.ts]:
```suggestion
	referencePosition?: Vector3;
```

**Comment by Mugen87** [comment]
Many thanks 👍 ! I've implemented your feedback.

**Review by Methuselah96 — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**